src/requests/textdocument.jl

```diff
@@ -70,6 +70,12 @@ JSONRPC.parse_params(::Type{Val{Symbol("textDocument/didSave")}}, params) = DidS
 function process(r::JSONRPC.Request{Val{Symbol("textDocument/didSave")},DidSaveTextDocumentParams}, server)
     uri = r.params.textDocument.uri
     doc = server.documents[URI2(uri)]
+    if r.params.text isa String
+        if get_text(doc) != r.params.text
+            @warn "Mismatch between server and client text."
+            set_text!(doc, r.params.text)
```

Comment:
I would just crash here, this shouldn't happen, right? And then we get a crash report for sure.